pandas/core/arrays/datetimelike.py

```diff
@@ -840,15 +843,6 @@ def searchsorted(self, value, side="left", sorter=None):
         elif isinstance(value, self._recognized_scalars):
             value = self._scalar_type(value)
 
-        elif isinstance(value, np.ndarray):
```

Comment:
I recall this was for performance, IIRC @topper-123 

can you run the benchamarks for searchsorted and report.

Reply:
Not sure if these touch searchsorted for datetimelike, but none of the searchsorted benchmarks changed:

```
(pandas-dev) danielsaxton> asv continuous -f 1.1 upstream/master searchsorted-date -b ^series_methods.SearchSorted
· Creating environments
· Discovering benchmarks
·· Uninstalling from conda-py3.6-Cython-matplotlib-numba-numexpr-numpy-odfpy-openpyxl-pytables-pytest-scipy-sqlalchemy-xlrd-xlsxwriter-xlwt.
·· Building 615147eb <searchsorted-date> for conda-py3.6-Cython-matplotlib-numba-numexpr-numpy-odfpy-openpyxl-pytables-pytest-scipy-sqlalchemy-xlrd-xlsxwriter-xlwt..................................
·· Installing 615147eb <searchsorted-date> into conda-py3.6-Cython-matplotlib-numba-numexpr-numpy-odfpy-openpyxl-pytables-pytest-scipy-sqlalchemy-xlrd-xlsxwriter-xlwt..
· Running 2 total benchmarks (2 commits * 1 environments * 1 benchmarks)
[  0.00%] · For pandas commit 80078ac0 <searchsorted-date^2> (round 1/2):
[  0.00%] ·· Building for conda-py3.6-Cython-matplotlib-numba-numexpr-numpy-odfpy-openpyxl-pytables-pytest-scipy-sqlalchemy-xlrd-xlsxwriter-xlwt....................................
[  0.00%] ·· Benchmarking conda-py3.6-Cython-matplotlib-numba-numexpr-numpy-odfpy-openpyxl-pytables-pytest-scipy-sqlalchemy-xlrd-xlsxwriter-xlwt
[ 25.00%] ··· Running (series_methods.SearchSorted.time_searchsorted--).
[ 25.00%] · For pandas commit 615147eb <searchsorted-date> (round 1/2):
[ 25.00%] ·· Building for conda-py3.6-Cython-matplotlib-numba-numexpr-numpy-odfpy-openpyxl-pytables-pytest-scipy-sqlalchemy-xlrd-xlsxwriter-xlwt...
[ 25.00%] ·· Benchmarking conda-py3.6-Cython-matplotlib-numba-numexpr-numpy-odfpy-openpyxl-pytables-pytest-scipy-sqlalchemy-xlrd-xlsxwriter-xlwt
[ 50.00%] ··· Running (series_methods.SearchSorted.time_searchsorted--).
[ 50.00%] · For pandas commit 615147eb <searchsorted-date> (round 2/2):
[ 50.00%] ·· Benchmarking conda-py3.6-Cython-matplotlib-numba-numexpr-numpy-odfpy-openpyxl-pytables-pytest-scipy-sqlalchemy-xlrd-xlsxwriter-xlwt
[ 75.00%] ··· series_methods.SearchSorted.time_searchsorted                                                                                                                                                                             ok
[ 75.00%] ··· ========= ============
                dtype               
              --------- ------------
                 int8    17.4±0.8μs 
                int16    17.5±0.2μs 
                int32    18.0±0.6μs 
                int64     19.0±2μs  
                uint8    16.6±0.9μs 
                uint16   17.7±0.7μs 
                uint32    16.8±1μs  
                uint64   18.4±0.3μs 
               float16    734±30μs  
               float32    120±9μs   
               float64   8.15±0.3μs 
                 str     6.75±0.3μs 
              ========= ============

[ 75.00%] · For pandas commit 80078ac0 <searchsorted-date^2> (round 2/2):
[ 75.00%] ·· Building for conda-py3.6-Cython-matplotlib-numba-numexpr-numpy-odfpy-openpyxl-pytables-pytest-scipy-sqlalchemy-xlrd-xlsxwriter-xlwt...
[ 75.00%] ·· Benchmarking conda-py3.6-Cython-matplotlib-numba-numexpr-numpy-odfpy-openpyxl-pytables-pytest-scipy-sqlalchemy-xlrd-xlsxwriter-xlwt
[100.00%] ··· series_methods.SearchSorted.time_searchsorted                                                                                                                                                                             ok
[100.00%] ··· ========= ============
                dtype               
              --------- ------------
                 int8     16.8±2μs  
                int16     19.3±5μs  
                int32     18.9±4μs  
                int64     17.4±2μs  
                uint8     17.9±2μs  
                uint16    17.0±1μs  
                uint32   17.1±0.4μs 
                uint64   18.6±0.2μs 
               float16    724±30μs  
               float32    111±2μs   
               float64   8.11±0.1μs 
                 str     6.56±0.2μs 
              ========= ============


BENCHMARKS NOT SIGNIFICANTLY CHANGED.
(pandas-dev) danielsaxton> asv continuous -f 1.1 upstream/master searchsorted-date -b ^categoricals.SearchSorted
· Creating environments
· Discovering benchmarks
·· Uninstalling from conda-py3.6-Cython-matplotlib-numba-numexpr-numpy-odfpy-openpyxl-pytables-pytest-scipy-sqlalchemy-xlrd-xlsxwriter-xlwt.
·· Installing 615147eb <searchsorted-date> into conda-py3.6-Cython-matplotlib-numba-numexpr-numpy-odfpy-openpyxl-pytables-pytest-scipy-sqlalchemy-xlrd-xlsxwriter-xlwt..
· Running 4 total benchmarks (2 commits * 1 environments * 2 benchmarks)
[  0.00%] · For pandas commit 80078ac0 <searchsorted-date^2> (round 1/2):
[  0.00%] ·· Building for conda-py3.6-Cython-matplotlib-numba-numexpr-numpy-odfpy-openpyxl-pytables-pytest-scipy-sqlalchemy-xlrd-xlsxwriter-xlwt...
[  0.00%] ·· Benchmarking conda-py3.6-Cython-matplotlib-numba-numexpr-numpy-odfpy-openpyxl-pytables-pytest-scipy-sqlalchemy-xlrd-xlsxwriter-xlwt
[ 12.50%] ··· Running (categoricals.SearchSorted.time_categorical_contains--)..
[ 25.00%] · For pandas commit 615147eb <searchsorted-date> (round 1/2):
[ 25.00%] ·· Building for conda-py3.6-Cython-matplotlib-numba-numexpr-numpy-odfpy-openpyxl-pytables-pytest-scipy-sqlalchemy-xlrd-xlsxwriter-xlwt...
[ 25.00%] ·· Benchmarking conda-py3.6-Cython-matplotlib-numba-numexpr-numpy-odfpy-openpyxl-pytables-pytest-scipy-sqlalchemy-xlrd-xlsxwriter-xlwt
[ 37.50%] ··· Running (categoricals.SearchSorted.time_categorical_contains--)..
[ 50.00%] · For pandas commit 615147eb <searchsorted-date> (round 2/2):
[ 50.00%] ·· Benchmarking conda-py3.6-Cython-matplotlib-numba-numexpr-numpy-odfpy-openpyxl-pytables-pytest-scipy-sqlalchemy-xlrd-xlsxwriter-xlwt
[ 62.50%] ··· categoricals.SearchSorted.time_categorical_contains                                                                                                                                                               5.89±0.3μs
[ 75.00%] ··· categoricals.SearchSorted.time_categorical_index_contains                                                                                                                                                         8.49±0.7μs
[ 75.00%] · For pandas commit 80078ac0 <searchsorted-date^2> (round 2/2):
[ 75.00%] ·· Building for conda-py3.6-Cython-matplotlib-numba-numexpr-numpy-odfpy-openpyxl-pytables-pytest-scipy-sqlalchemy-xlrd-xlsxwriter-xlwt...
[ 75.00%] ·· Benchmarking conda-py3.6-Cython-matplotlib-numba-numexpr-numpy-odfpy-openpyxl-pytables-pytest-scipy-sqlalchemy-xlrd-xlsxwriter-xlwt
[ 87.50%] ··· categoricals.SearchSorted.time_categorical_contains                                                                                                                                                               6.00±0.3μs
[100.00%] ··· categoricals.SearchSorted.time_categorical_index_contains                                                                                                                                                         6.63±0.2μs

BENCHMARKS NOT SIGNIFICANTLY CHANGED.
```

Reply:
I don't recall having touched this method, so probably ok